bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaWaterHeatingSystemHandler.java

```diff
@@ -0,0 +1,225 @@
+        // override state type because the cloud sends consumption in percent
+        cacheStateType(POWER_HEAT_ELEC_STATE, TYPE_DECIMAL);
+    }
+
+    @Override
+    public void updateThingChannels(SomfyTahomaState state) {
+        if (OPERATING_MODE_STATE.equals(state.getName()) && state.getValue() instanceof Map) {
+            logger.debug("Operating Mode State: {}  {}", state.getValue().getClass().getName(), state.getValue());
+
+            Map<String, String> data = (Map<String, String>) state.getValue();
+
+            Object relaunchValue = data.get("relaunch");
+            if (relaunchValue != null) {
+                this.boostMode = relaunchValue.toString().equalsIgnoreCase("on");
+                logger.debug("Boost Value: {}", this.boostMode);
+                updateState(BOOST_MODE, OnOffType.from(this.boostMode));
+            }
+
+            Object awayValue = data.get("absence");
+            if (awayValue != null) {
+                this.awayMode = awayValue.toString().equalsIgnoreCase("on");
+                logger.debug("Away Value: {}", this.awayMode);
+                updateState(AWAY_MODE, OnOffType.from(this.boostMode));
+            }
+        } else if (TARGET_TEMPERATURE_STATE.equals(state.getName())) {
+            logger.debug("Target Temperature: {}", state.getValue());
+            // 50 -> 3
+            // 54.5 -> 4
+            // 62 -> 5
+            Double temp = null;
+            try {
+                temp = Double.parseDouble(state.getValue().toString());
+            } catch (NumberFormatException e) {
+                logger.warn("Unexcepted pre-defined value for Target State Temperature: {}", state.getValue());
+                return;
+            }
+            int v = 0;
+            if (temp == 50) {
+                v = 3;
+            } else if (temp == 54.5) {
+                v = 4;
+            } else if (temp == 62) {
+                v = 5;
+            }
+
+            updateState(SHOWERS, new DecimalType(v));
+        }
+
+        super.updateThingChannels(state);
+    }
+
+    private void sendOperatingMode() {
+        sendCommand(COMMAND_SET_CURRENT_OPERATING_MODE, String.format("[ { \"relaunch\":\"%s\", \"absence\":\"%s\"} ]",
+                (this.boostMode ? "on" : "off"), (this.awayMode ? "on" : "off")));
+    }
+
+    private void sendBoostDuration(int duration) {
+        sendCommand(COMMAND_SET_BOOST_MODE_DURATION, "[ " + duration + " ]");
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        super.handleCommand(channelUID, command);
+
+        if (command instanceof RefreshType) {
+            return;
+        } else {
+            logger.debug("Command received: {}/{}", channelUID.getId(), command.toString());
+
+            if (BOOST_MODE_DURATION.equals(channelUID.getId())) {
+                int duration = 0;
+                try {
+                    Integer.parseInt(command.toString());
```

Comment:
You must assign the result to `duration` !

Reply:
well spotted again!